woocommerce/woocommerce-android
# Shipping Labels: add ability to move items to new packages

This handles the first task of #4002, by allowing moving items to new packages, the PR is bigger than I planned.

When reviewing, please note the following:
1. The commits 50be635, 868c6d8, 1f8286e, and 6ca704d were a preparation to allow having a consistent identification of packages (both their name and `packageId`) across different steps, and it will be used also when trying to move items to existing packages.
2. I didn't add any unit tests for the new functionality, it'll be done in a separate PR.

<img width=360 src="https://user-images.githubusercontent.com/1657201/124125324-64121b80-da71-11eb-92fc-6df24c8f3c78.gif" />


#### Testing
1. Create an order eligible for shipping label creation with multiple items.
2. Open the order in the app.
3. Click on Create shipping label button.
4. Pass the origin and destination steps.
5. Click on Continue in the packaging step.
6. Click on Move button in one of the items.
7. Notice that the Move Item dialog is displayed.
8. Select "Add to New Package" option.
9. Click on "Move"
10. Confirm that the item has been moved to a different package.
11. Confirm that the new package is expanded, and the other packages are collapsed.


Update release notes:
